src/Firestore/DocumentReference.php

```diff
@@ -0,0 +1,374 @@
+     * $document->update([
+     *     'country' => FirestoreClient::DELETE_FIELD
+     * ]);
+     * ```
+     *
+     * ```
+     * // Documents can be updated using field paths as well.
+     * $document->update([
+     *     'name' => 'John',
+     *     'country' => 'USA',
+     *     'cryptoCurrencies.bitcoin' => 0.5,
+     *     'cryptoCurrencies.ethereum' => 10,
+     *     'cryptoCurrencies.litecoin' => 5.51
+     * ]);
+     * ```
+     *
+     * @param array $fields An array containing field names paired with their value.
+     *        Accepts a nested array, or a simple array of field paths.
+     * @param array $options {
+     *     Configuration Options
+     *
+     *     @type array $precondition An optional precondition on the document. If
+     *           this is set and not met by the target document, the write will
+     *           fail. Allowed arguments are `(bool) $exists` and
+     *           {@see Google\Cloud\Core\Timestamp} `$updateTime`.
+     *           To completely disable precondition checks, provide an empty array
+     *           as the value of `$precondition`. **Defaults to**
+     *           `['exists' => true]` (i.e. Document must exist in Firestore).
+     * }
+     * @return array
+     * @throws ConflictException If the
+     *         precondition is not met.
+     */
+    public function update(array $fields, array $options = [])
+    {
+        $writer = $this->batchFactory();
+
+        $writer->update($this->name, $fields, $options);
+
+        return $writer->commit($options);
+    }
+
+    /**
+     * Delete a Firestore document.
+     *
+     * Example:
+     * ```
+     * $document->delete();
+     * ```
+     *
+     * @param array $options {
+     *     Configuration Options
+     *
+     *     @type array $precondition An optional precondition on the document. If
+     *           this is set and not met by the target document, the write will
+     *           fail. Allowed arguments are `(bool) $exists` and
+     *           {@see Google\Cloud\Core\Timestamp} `$updateTime`.
+     *           To completely disable precondition checks, provide an empty array
+     *           as the value of `$precondition`. **Defaults to**
+     *           `['exists' => true]` (i.e. Document must exist in Firestore).
+     * }
+     * @return array
+     * @throws ConflictException If the precondition is not met.
+     */
+    public function delete(array $options = [])
+    {
+        $writer = $this->batchFactory();
+
+        $writer->delete($this->name, $options);
+
+        return $writer->commit($options);
+    }
+
+    /**
+     * Get a read-only snapshot of the document.
+     *
+     * Example:
+     * ```
+     * $snapshot = $document->snapshot();
+     * ```
+     *
+     * @param array $options {
+     *     Configuration Options
+     *
+     *     @type array $mask A list of fields to return. If not set, returns all
```

Comment:
The way we build requests, it's actually zero extra work for us to support the full [request options](https://cloud.google.com/firestore/docs/reference/rpc/google.firestore.v1beta1#batchgetdocumentsrequest) for this request. Since they're supported, I figured documenting them would be preferred for users who might have use for them.

Reply:
I would prefer to remove this documentation. It breaks one of the goals we want to achieve with the Firestore SDKs and publicizes API surface area that is only available in 1 of our 7 Server SDKs. Things you get for free in PHP might take a lot of time to implement in other languages and we already have very vocal customer pointing out missing features in some of our SDKs (there is demand for select() in the mobile SDKs, and listen() in the non-Node Server SDKs).

As mentioned above, there is no need to break this feature. But treating it more like an easter egg would be preferred from my end.